File: 0. Project 심다/simda/src/main/java/ssafy/a709/domain/Chatroom.java
package ssafy.a709.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import ssafy.a709.dto.ChatRoomDTO;

import javax.persistence.*;
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
public class Chatroom {
    public static Chatroom chageToChatroom(ChatRoomDTO chatRoomDTO){
        return Chatroom.builder()
                .chatroomId(chatRoomDTO.getChatRoomId())
                .user1(User.changeToUser(chatRoomDTO.getUser1()))
                .user2(User.changeToUser(chatRoomDTO.getUser2()))
                .chat(Chat.chageToChat(chatRoomDTO.getLChat()))
                .build();
    };



    // Chatroom Id
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "chatroom_id")
    private int chatroomId;

    // Many(User1 Id) to One(User Id)
    @ManyToOne
    @JoinColumn(name = "user1_id", referencedColumnName = "user_id")
    private User user1;

    // Many(User2 Id) to One(User Id)
    @ManyToOne
    @JoinColumn(name = "user2_id", referencedColumnName = "user_id")
    private User user2;

    // One(Last Chat Id) to One(Chat Id)
    @OneToOne
    @JoinColumn(name = "l_chat_id", referencedColumnName = "chat_id")
    private Chat chat;
}

File: frontend/simda/lib/models/ChatRoom.dart
import 'package:simda/models/Chat.dart';
import 'package:simda/models/User.dart';

class ChatRoom {
  final int chatRoomId;
  final Chat lchat;
  final User user1;
  final User user2;

  ChatRoom({
    required this.chatRoomId,
    required this.lchat,
    required this.user1,
    required this.user2,
  });

  factory ChatRoom.fromJson(Map<String, dynamic> json) {
    return ChatRoom(
      chatRoomId: json['chatRoomId'],
      lchat: json['lchat'],
      user1: json['user1'],
      user2: json['user2'],
    );
  }
}
